test/shared/src/main/scala/zio/test/Spec.scala

```diff
@@ -227,34 +227,36 @@ final case class Spec[-R, +E, +L, +T](caseValue: SpecCase[R, E, L, T, Spec[R, E,
     }
 
   /**
-   * Uses the specified `Managed` to provide each test in this spec with its
+   * Uses the specified `ZManaged` to provide each test in this spec with its
    * required environment.
    */
-  final def provideManaged[E1 >: E](managed: Managed[E1, R])(implicit ev: NeedsEnv[R]): Spec[Any, E1, L, T] =
-    transform[Any, E1, L, T] {
-      case SuiteCase(label, specs, exec) => SuiteCase(label, specs.provideManaged(managed), exec)
-      case TestCase(label, test)         => TestCase(label, test.provideManaged(managed))
+  final def provideManaged[R0, E1 >: E](managed: ZManaged[R0, E1, R])(implicit ev: NeedsEnv[R]): Spec[R0, E1, L, T] =
```

Comment:
I'd be tempted to leave the old one, and have it delegate to a new `provideSomeManaged`, just for consistency with ZIO.

Reply:
Okay. Do you want to do the same thing with `provideSomeShared` and have it delegate to `provideSomeManagedShared`? The only reason I did it this was was `provideSomeManagedShared` seemed really long and then I wanted to be consistent within Spec.

Reply:
Right, I think ideally we'd achieve consistency within spec, and across Spec and ZIO. Your call though!

Another idea is introducing shorter aliases, both here and within ZIO. From my POV, it's really the `Shared` that pushes it over the edge into `almostUnreasonableMethodNameLength`. 😄 

But of course we need a shared variant for these things or a way to do it.

Reply:
Do you have an idea for what you would like the shorter aliases to be? Happy to add those across core, streams, and tests.